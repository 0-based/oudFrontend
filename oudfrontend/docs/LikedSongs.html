<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: LikedSongs</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: LikedSongs</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>LikedSongs<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">this is a component that renders likedSongs page</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="LikedSongs"><span class="type-signature"></span>new LikedSongs<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>state</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last">carries the state of the component
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>recived</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">true if the data of the playlist is fetched correctly false otherwise</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>items</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last">carries all the information of the playlist</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>tracks</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;track></span>


            
            </td>

            

            

            <td class="description last">array of all the songs in the playlist</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    

        <tr>
            
                <td class="name"><code>playing</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">true when the playist is playing. Otherwise, it is false</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>queued</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">true when the playist is added to queue. Otherwise, it is false</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>clikedID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">this is used to mark all songs as unclicked</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    
    <dt class="tag-author">Author:</dt>
    <dd class="tag-author">
        <ul>
            <li><a href="mailto:ahmedwa1999@gmail.com">Ahmed Walid</a></li>
        </ul>
    </dd>
    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="likedSongs_likedSongs.jsx.html">likedSongs/likedSongs.jsx</a>, <a href="likedSongs_likedSongs.jsx.html#line38">line 38</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        


    





    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="addToQueue"><span class="type-signature"></span>addToQueue<span class="signature">(tracks, length)</span><span class="type-signature"> &rarr; {void}</span></h4>
    

    



<div class="description">
    add the tracks to queue and resume the player
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>tracks</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;track></span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>length</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="likedSongs_likedSongs.jsx.html">likedSongs/likedSongs.jsx</a>, <a href="likedSongs_likedSongs.jsx.html#line61">line 61</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">void</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id="componentDidMount"><span class="type-signature"></span>componentDidMount<span class="signature">()</span><span class="type-signature"> &rarr; {void}</span></h4>
    

    



<div class="description">
    It fetches the data of the playlist from the database and checks if it exists in the likedPlaylists table
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="likedSongs_likedSongs.jsx.html">likedSongs/likedSongs.jsx</a>, <a href="likedSongs_likedSongs.jsx.html#line109">line 109</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">void</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id="markAllUnclicked"><span class="type-signature"></span>markAllUnclicked<span class="signature">()</span><span class="type-signature"> &rarr; {void}</span></h4>
    

    



<div class="description">
    it changes the state so that all song will be marked as unclicked
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="likedSongs_likedSongs.jsx.html">likedSongs/likedSongs.jsx</a>, <a href="likedSongs_likedSongs.jsx.html#line135">line 135</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">void</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id="pause"><span class="type-signature"></span>pause<span class="signature">()</span><span class="type-signature"> &rarr; {void}</span></h4>
    

    



<div class="description">
    pauses the player
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="likedSongs_likedSongs.jsx.html">likedSongs/likedSongs.jsx</a>, <a href="likedSongs_likedSongs.jsx.html#line90">line 90</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">void</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id="playButtonClicked"><span class="type-signature"></span>playButtonClicked<span class="signature">()</span><span class="type-signature"> &rarr; {void}</span></h4>
    

    



<div class="description">
    Called Whenever the user clicked on the PLAY button and it adds all the songs of the playlist to the queue by a post request
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="likedSongs_likedSongs.jsx.html">likedSongs/likedSongs.jsx</a>, <a href="likedSongs_likedSongs.jsx.html#line71">line 71</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">void</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id="resume"><span class="type-signature"></span>resume<span class="signature">()</span><span class="type-signature"> &rarr; {void}</span></h4>
    

    



<div class="description">
    resums the player
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="likedSongs_likedSongs.jsx.html">likedSongs/likedSongs.jsx</a>, <a href="likedSongs_likedSongs.jsx.html#line99">line 99</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">void</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="addToPlaylist.html">addToPlaylist</a></li><li><a href="Album.html">Album</a></li><li><a href="CreatePlaylist.html">CreatePlaylist</a></li><li><a href="LikedSongs.html">LikedSongs</a></li><li><a href="Playlist.html">Playlist</a></li><li><a href="Song.html">Song</a></li><li><a href="SongList.html">SongList</a></li></ul><h3>Global</h3><ul><li><a href="global.html#addToQueue">addToQueue</a></li><li><a href="global.html#HeaderBody">HeaderBody</a></li><li><a href="global.html#HeaderBodyBottom">HeaderBodyBottom</a></li><li><a href="global.html#HeaderBodyTop">HeaderBodyTop</a></li><li><a href="global.html#pause">pause</a></li><li><a href="global.html#resume">resume</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Thu Apr 02 2020 22:33:34 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>