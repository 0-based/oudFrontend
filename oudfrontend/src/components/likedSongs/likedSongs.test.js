import React from 'React'
import renderer from 'react-test-renderer';
import LikedSongs from './likedSongs';
import Enzyme, {shallow} from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new EnzymeAdapter()});


const setup = () =>{
    return shallow(<LikedSongs.WrappedComponent/>);
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-testid="${val}"]`);
}
describe('album Component', ()=>{
        
    describe('renders correctly', ()=>{
        let component;
        beforeEach (()=>{
            component = setup();
        })
        it('renders album component', ()=>{
            const wrapper = findByTestAttr(component, "wrapper");
            expect(wrapper.length).toBe(1);
        });
        it('renders playlistHeader component', ()=>{
            const wrapper = findByTestAttr(component, "sidebar");
            expect(wrapper.length).toBe(1);
        });
        it('renders album component', ()=>{
            const wrapper = findByTestAttr(component, "likedSongs");
            expect(wrapper.length).toBe(1);
        });
        it('renders playlistIamge component', ()=>{
            const wrapper = findByTestAttr(component, "playlistIamge");
            expect(wrapper.length).toBe(1);
        });
        it('renders playlistIamgeContainer component', ()=>{
            const wrapper = findByTestAttr(component, "playlistIamgeContainer");
            expect(wrapper.length).toBe(1);
        });
        it('renders playlistHeaderBody component', ()=>{
            const wrapper = findByTestAttr(component, "playlistHeaderBody");
            expect(wrapper.length).toBe(1);
        });
        
        it('renders songList component', ()=>{
            const wrapper = findByTestAttr(component, "songList");
            expect(wrapper.length).toBe(1);
        });
    
    })
    
   /* describe('snapshot test', ()=>{
        it('renders correctly', () => {
            const tree = renderer
              .create(<LikedSongs.WrappedComponent/>)
              .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });*/
})