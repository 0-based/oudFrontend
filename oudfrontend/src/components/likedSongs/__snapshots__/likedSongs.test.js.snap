// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`album Component snapshot test for the playlist renders correctly 1`] = `
<div
  className="playlist"
  data-testid="likedSongs"
>
  <div
    className="row"
  >
    <div
      className="playlistHeader row col-xs-4 col-md-6 col-lg-4 col-xl-4"
      data-testid="playlistHeader"
      onClick={[Function]}
    >
      <div
        className="playlistImageContainer col col-lg-12 col-md-12 col-sm-4 col-xs-4"
        data-testid="playlistIamgeContainer"
      >
        <img
          alt="playlist img"
          className="playlistImage"
          data-testid="playlistIamge"
          src="https://t.scdn.co/images/3099b3803ad9496896c43f22fe9be8c4.png"
        />
      </div>
      <div
        className="playlistHeaderBody col col-lg-12 col-md-12 col-sm-8 col-xs-8"
        data-testid="playlistHeaderBody"
      >
        <div
          data-testid="HeaderBody"
        >
          <h2
            className="whiteText"
            data-testid="title"
          >
            Liked Songs
          </h2>
          <button
            className="playButton"
            data-testid="playButton"
            onClick={[Function]}
            variant="outline-success"
          >
            PLAY
          </button>
          <p>
            <span
              data-testid="songsNumber"
            >
              0
               
            </span>
            <span
              data-testid="songsLiteral"
            >
              song
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      className="col-xs-8 col-md-6 col-lg-8 col-xl-8"
      data-testid="songsList"
    >
      <h1
        data-testid="loading"
      >
        LOADING ...
      </h1>
    </div>
  </div>
</div>
`;
